lib/rucio/daemons/hermes/hermes.py

```diff
@@ -240,6 +240,45 @@ def aggregate_to_influx(
     return 204
 
 
+def build_message_dict(
+        bulk: int,
+        worker_number: int,
+        total_workers: int,
+        message_dict: dict[str, list[dict[str, Any]]],
+        logger: "LoggerFunction",
+        service: str = "",
+) -> None:
+    """
+    Build a dictionary with the keys being the services, and the values a list of the messages (built up of dictionary / json information)
+
+    :param bulk                Intiger for number of messages to retrieve, default behaviour is total, query_by_service is per service
+    :param worker_number       Passed to thread in retrieve_mesasges for Identifier of the caller thread as an integer.
+    :param totoal_workers      Passed to total_threads for Maximum number of threads as an integer.
+    :param message_dict        Either empty dictionary to be build, or build upon when using query_by_service
+    :param logger:             The logger object.
+    :param service             When passed queries the message table for bulk number for that specific service.
+
+    :returns None, but builds on the dictionary message_dict passed to this fuction (for when querying multiple services).
+    """
+    start_time = time.time()
+    messages = retrieve_messages(
+        bulk=bulk,
+        old_mode=False,
+        thread=worker_number,
+        total_threads=total_workers,
+        service_filter=service,
+    )
+
+    if messages and service not in message_dict:
```

Comment:
This is related to the above comment about changing the default value of `service` to `None` (i.e. `service: Optional[str] = None,`)

If you were to use the default value of `''` as `service`, then when `query_by_service` is false, message_dict would just enter all messages retrieved under the key `''`.  This breaks the logic that comes after line 367 in the `hermes` function - it looks for services as the keys in `message_dict`, but it would not be able to find them.